protobuf/lib/src/protobuf/json/json_web.dart

```diff
@@ -0,0 +1,463 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert' show base64Decode, base64Encode;
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+
+import 'package:fixnum/fixnum.dart' show Int64;
+
+import '../consts.dart';
+import '../internal.dart';
+import '../utils.dart';
+
+@JS('JSON')
```

Comment:
I don't think it's a big deal, but these extension types and external method means that `json_web.dart` has public API which is a strict superset of `json_vm.dart`. When using config specific imports do try to keep the APIs for each implementation of the library the same.